### PR TITLE
Add progress bar for EME

### PR DESCRIPTION
**Pull request by caseyflex**
Together with the backend PR, this adds a progress bar for EME simulations.

### CODE REVIEW

File: tidy3d/web/api/webapi.py [hunk 1]
```diff
@@ -436,32 +436,49 @@ def monitor_preprocess() -> None:
...
+                ):
+                    perc_done, _ = get_run_info(task_id)
+                    new_description = "solver progress"
+                    progress.update(pbar_pd, completed=perc_done, description=new_description)
+                    time.sleep(RUN_REFRESH_TIME)
+
+                perc_done, _ = get_run_info(task_id)
+                new_description = "solver progress"
+                progress.update(
+                    pbar_pd, completed=perc_done, refresh=True, description=new_description
```
Comment (momchil-flex) [inline tidy3d/web/api/webapi.py]: In the FDTD case, we set `completed=100` here to force the progress bar to look "done" once the solver is done. Any reason you set `perc_done` instead? Which I think should have no effect then?

**Review by momchil-flex — APPROVED** [review]
Thanks, this is great! In my tests it looked a little uneven in the progress (I mean a bit uneven time between updates) but it's nice to have *something* happening. :)

File: tidy3d/web/api/webapi.py (hunk 1 above)
Comment (caseyflex) [reply tidy3d/web/api/webapi.py]: thanks for the catch!

**Comment by caseyflex** [comment]
> Thanks, this is great! In my tests it looked a little uneven in the progress (I mean a bit uneven time between updates) but it's nice to have _something_ happening. :)

Yeah, it's definitely not entirely even, just based on counting steps, but should be better than nothing. We can update it to use the time estimation later.

**Pull request merged**